packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -957,3 +962,22 @@ function isArrayLengthExpression(
   const objectType = getConstrainedTypeAtLocation(services, node.object);
   return isTypeArrayTypeOrUnionOfArrayTypes(objectType, typeChecker);
 }
+
+/**
+ * Verify is the type is a branded boolean (e.g. `type Foo = boolean & { __brand: 'Foo' }`)
+ *
+ * @param type The type checked
+ */
+function isBrandedBoolean(type: ts.Type): boolean {
+  if (!tsutils.isTypeFlagSet(type, ts.TypeFlags.Intersection)) {
+    return false;
+  }
+
+  const types = (type as ts.IntersectionType).types;
+  return types.some(childType =>
+    tsutils.isTypeFlagSet(
+      childType,
+      ts.TypeFlags.BooleanLiteral | ts.TypeFlags.Boolean,
+    ),
+  );
```

Comment:
IIRC, you can do:

```suggestion
  return type.isIntersection() && 
	  type.types.some(childType =>
	    tsutils.isTypeFlagSet(
	      childType,
	      ts.TypeFlags.BooleanLiteral | ts.TypeFlags.Boolean,
	    ),
	  );
```

and then you won't need the type assertion

Reply:
👍 yup, this. Searching for `isIntersection` shows a couple of options.